tests/CollectionTest.php

```diff
@@ -2062,6 +2066,43 @@ public function testIsVersioningEnabled()
         ));
         $this->assertFalse($this->database->col3->isVersioningEnabled());
     }
+
+    /**
+     * @throws Exception
+     * @throws \MongoException
+     */
+    public function testRenameNonExistentCollection()
+    {
+        $this->expectException(Exception::class);
+        $this->expectExceptionCode(26);
+        $this->expectExceptionMessage('Error: #26: source namespace does not exist');
+
+        // set test collection
+        $this->collection = $this->database->getCollection('test');
+
+        // rename non-existent collection to cause a MongoClient error
+        $this->collection->renameCollection('phpmongo_test_collection', true);
+    }
+
+    /**
+     * @throws Exception
+     * @throws \MongoException
+     */
+    public function testRenameExistentCollection()
+    {
+        $sourceCollection = 'source_collection';
+
+        // creating fake-collection to rename it then in test db context
+        $this->database->createCollection($sourceCollection);
+
+        $this->collection = $this->database->getCollection($sourceCollection);
+
+        // rename an existent collection
+        $status = $this->collection->renameCollection('phpmongo_test_collection', true);
```

Comment:
1) add test with trying to stopre document to renamed collection.
2) renamed collection return old name?

Reply:
1. Ok
2. I think it will be better to return collection, to let developers operate with new one.